### PR TITLE
Add doc resurrection tests

### LINKED ISSUE #1131: XATTRs: Test doc resurrection with SDK interop
- [ ] Delete / Create through SDK, with a delay
- [ ] Tombstone / Resurrect through Sync Gateway
- [ ] Sync Gateway monitor expected revs for SDK Delete (wait for this) / Create doc (wait for this)
- [ ] SDK should see of not see the doc

### PR DESCRIPTION
**Pull request by sethrosetter**
#### Fixes #1131 #1246 .

- [x] Ran `flake8`
- [x] Ran `run_repo_tests.sh`

#### Changes proposed in this pull request:

- Add scenarios for doc resurrection (tombstone / purge)
- Rename base docker image
- Reduce RBAC permissions to sync gateway minimum permissions

### CODE REVIEW

File: testsuites/syncgateway/functional/tests/test_crud.py [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+    - Create docs (set B) via SDK
+    - Delete SDK docs (set B) via Sync Gateway
+    - Delete SG docs (set A) via SDK
+    - Verify docs (set B) are deleted via Sync Gateway
+    - Verify docs (set B) are deleted via SDK
+    - Verify docs (set A) are deleted via Sync Gateway
+    - Verify docs (set A) are deleted via SDK
+    - Create docs (set A) via Sync Gateway
+    - Create docs (set B) via SDK
+    - Verify revs (set A, B) are generation 3 via Sync Gateway
```
Comment (adamcfraser) [inline testsuites/syncgateway/functional/tests/test_crud.py]: I believe that SDK docs are going to be recreated as generation 1.  The SDK create on line 52 doesn't preserve any existing xattrs, and so will be imported as a new rev 1.

File: testsuites/syncgateway/functional/tests/test_crud.py [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+
+    XATTRs / purge
```
Comment (adamcfraser) [inline testsuites/syncgateway/functional/tests/test_crud.py]: Is there a difference between the tombstone and purge scenario from the SDK's perspective?

File: testsuites/syncgateway/functional/tests/test_crud.py (hunk 1 above)
Comment (sethrosetter) [inline testsuites/syncgateway/functional/tests/test_crud.py]: Ok. I will re-review.

File: testsuites/syncgateway/functional/tests/test_crud.py (hunk 2 above)
Comment (sethrosetter) [inline testsuites/syncgateway/functional/tests/test_crud.py]: No. I have updated the logic.

**Comment by sethrosetter** [comment]
@raghusarangapani This should be good now. All test pass so merging this should not cause any noise.

File: testsuites/syncgateway/functional/tests/test_crud.py (hunk 1 above)
Comment (sethrosetter) [inline testsuites/syncgateway/functional/tests/test_crud.py]: The verification logic has been updated.

**Comment by raghusarangapani** [comment]
looks good

**Pull request merged**